### DISCUSSION TITLE
Support for JSX 4

### DISCUSSION THREAD

**Discussion by chriswilty** [giraud/reasonml-idea-plugin#394] [Feature request]
Now that rescript/react 0.11+ supports JSX v4, is there a plan to add support to this plugin? The bucklescript config has changed a bit, which may be why it doesn't "just work", if the plugin is collecting error info from rescript for display in the editor. But I'm new to this plugin and to ReScript in general, so I may well have missed something big 😅

**Comment by giraud** [comment]
yes there is a plan to support new JSX, but I was waiting new versions to be available in npm.
I am not aware that the new versions have been released.

This plugin is not using LSP, and is parsing source code directly. it uses rescript binary to compile (get errors), reformat and infer types. 
I tried to explain it here: https://giraud.github.io/reasonml-idea-plugin/docs/contributing/plugin-architecture .

Unfortunately the Rescript team is not releasing stable binaries and code must be adapted regularly. 
I didn’t took time to check the updates yet.

**Comment by giraud** [comment]
jsx4 and rescript.json are supported starting with version 0.118
